packages/cli/src/ui/components/HistoryItemDisplay.tsx

```diff
@@ -54,6 +57,17 @@ export const HistoryItemDisplay: React.FC<HistoryItemDisplayProps> = ({
   embeddedShellFocused,
   availableTerminalHeightGemini,
 }) => {
+  const config = useConfig();
+  const startTime = useRef(Date.now());
+
+  useEffect(() => {
```

Comment:
I'm skeptical this measures what we think it does. Also do not like that it adds an extra useEffect just to try to track this.  This will only update when config updates which is surely not every frame.
The right approach is to send out a pr to ink to add callbacks to properly test this. 
DebugProfile gives some breadcrumbs on ways this could be done without Ink changes.